lib/util/json-writer/editor-config.ts

```diff
@@ -0,0 +1,42 @@
+import { KnownProps, parseSync } from 'editorconfig';
+import type { CodeFormat } from './code-format';
+import { IndentationType } from './indentation-type';
+
+export class EditorConfig {
+  public static getCodeFormat(fileName: string): CodeFormat {
+    const knownProps = parseSync(fileName);
```

Comment:
We don't like every using sync methods, can it be switched to async?

Reply:
And how far "up" does this method go - e.g. would it find an `.editorconfig` in the root directory if one existed?

Reply:
https://github.com/editorconfig/editorconfig-core-js#parsefilepath-options yeah, it should find and `.editorconfig` file in the root directory

Reply:
I have tried rewrite to async await, but by some reason unit tests fails, therefore I propose to go on with `parseSync`

Reply:
Please use async and push changes, we can then have a look why it fails.

Reply:
Done. Seems like content for .editorconfig file does not mock properly by some reason